### ISSUE TITLE
Weekly Community Check-In #90 - Modifying Expectations

### ISSUE THREAD

**Issue by noi5e** [publiclab/plots2#9112]
Hi everybody! 👏

We all at Public Lab 🎈 - learn, grow, work, brainstorm ideas, contribute together so why not share about our weekly goals and the awesome work we have done at Public Lab with each other, so we can support and collaborate with each other better. We have a Community Check-In each week, where every community member can share something about their work from the past week and about their current week's goal 🎯 . You are also welcome to share fun-fact 😄 , new ideas 💡 , your learning goals ☑️.

We believe in collaborative efforts to support our community. We are running a learning platform which helps a newcomer to become a leader of tomorrow. 💯

## Modifying Expectations
Hi everyone, I'm Will, one of the two [Outreachy](https://www.outreachy.org) Interns working with Public Lab! I'm working on refactoring Public Lab's Comment Editor.

I'm borrowing this week's theme of **Modifying Expectations** from the Outreachy organizers. Every two weeks, they send us a prompt to help us reflect on our internship:

> **Delays to projects happen.** Maybe your project turned out to be more complicated than you or your mentor anticipated. Maybe you needed to learn some concepts before you could tackle project tasks. Maybe the community documentation wasn't up-to-date or was wrong... Projects often seem simple until you start working on them. **Project timelines are usually a very optimistic view of what could happen if everything goes exactly as planned. It often doesn't, but people still make optimistic plans.**

Making optimistic plans even if you know something might take you a long time is called [the planning fallacy](https://en.wikipedia.org/wiki/Planning_fallacy).

If you feel called, leave a comment here and **tell us about a time when you had to modify your own expectations, or caught yourself making overly optimistic plans!** (I'll start by leaving a comment)

## This week's focus Issues

The [high priority issues](https://github.com/publiclab/plots2/labels/high-priority) this week include:
- #8864 (_Test Failures Caused by Revision Fixtures with the Same NID_)

We'd appreciate some help with this issue!

We will add a thank you message here in next week's check-in if you contribute to this one. Thanks in advance 😄

## Thank Yous

Thank you so much to @jywarren, @cesswairimu, @ebarry, @icarito who solved or contributed to discussions in some gnarly issues: #8864 #8177 #8858

Thank you to @RuthNjeri for providing code review in #9057

Thank you to the contributors who submitted PRs in the last week @gauravahlawat81 @harshdhonchak @Antonio-Riccelli @Manasa2850 @1liale @BenUnyolo (probably forgetting some people, let me know if I overlooked you!)

## Next Check-in
If you would like to open the next check-in.. Leave a comment below. We are happy to help if its your first time

**Thanks everyone for making Public Lab awesome** :heart:  
Have a great week ahead all :balloon:

**Comment by noi5e** [comment]
So the theme is about **times when we had to _Modify Our Expectations_**. I'll start.

I was just looking at the schedule that I made when I applied to be an Outreachy Intern with Public Lab:

1. Week 1: Write System Tests
2. Week 2: Write System Tests
3. Week 3: Find, Solve, and Write Up Critical Bugs
4. Week 4: Find, Solve, and Write Up Critical Bugs
5. Week 5: Image Upload
6. Week 6: Image Upload
7. Week 7: Rich-Text Entry
8. Week 8: Rich-Text Entry
9. Week 9: Save & Preview Text
10. Week 10: Save & Preview Text
11. Week 11: Overflow Week
12. Week 12: Overflow Week
13. Week 13: Overflow Week

### 😆 😆 😆 

Looking at this schedule made me chuckle a bit because I had no idea what I was getting into at the time. 

Here's how I've actually been spending my time:

1. Week 1: Write System Tests
2. Week 2: Write System Tests
3. Week 3: Write System Tests
4. Week 4: Write System Tests
5. Week 5: Image Upload Bugs _(Actually, I never stopped writing System Tests ✅ )_
6. Week 6: Image Upload Bugs
7. Week 7: Rich-Text Bugs
8. Week 8: Comment Editor Refactoring
9. Week 9: Comment Editor Refactoring
10. Weeks 10-12: 🚀 🚀 🚀 

Basically:
- I underestimated how much time I would be spending writing system tests.
- I also underestimated **how much I would learn** when writing tests! I didn't really have to spend two weeks 'hunting for bugs' like I thought I would.
- Solving the image upload bugs took about as much time as I thought it would (two weeks). 
  - But those two weeks felt LOOONG. Those were tricky bugs to solve.
- Once image upload bugs were solved, rich-text bugs went a LOT quicker than I thought. I didn't have to spend two whole weeks fixing them (more like 3 days).
- Similarly, I re-prioritized and am not going to spend two weeks working on Save & Recover issues! Instead, I've been doing a deep refactor of the Comment Editor.

Let us know about a time when you had to change plans!

### 🐉 🌟 🍭

**Comment by RuthNjeri** [comment]
haha @noi5e 3 overflow weeks? haha... I discovered along the way that there were small issues that came up in addition to the other issues that I needed to solve, also, writing up an FTO issue was not as simple as I thought it would be, because you need to write up an issue that makes it easy for a beginner to contribute. It was difficult breaking up an issue that touches multiple files which would make an FTO contributor have to deal with conflicts... but this is a good thing because I discovered that a lot of FTO contributors miss minor guidelines such as branching out and they also need quick documentation on how to solve conflicts... this would be a good issue to write up... and attach the document to the FTO template....

Also, the Outreachy blogs.... sometimes writing code seems like the priority but the blogs help in reflecting on the progress. I am planning on combining two blog prompts into one this week... I will probably post it next week though(rough estimate 😅 )

**Comment by Sagarpreet** [comment]
Hey @all,
Very good theme @noi5e 😄 Always add a buffer period in all your plans in case anything unexpected happens :P 

TODO this week: I plan to catch up with Outreachy interns first! Happy coding 🎉

**Comment by noi5e** [comment]
@RuthNjeri Thanks!!! I've also been wanting to come up with a good FTO issue. It's a bit of a struggle to pick a good one. I want it to be a cool learning experience, but if it brings up merge conflicts and failed tests, I know that it can be intimidating, and I might not have the time to coach them through it. But then again, if it wasn't for FTO issues, I wouldn't be here.

@Sagarpreet Thanks for the detailed reviews earlier!

**Comment by noi5e** [comment]
> @noi5e there was a discussion of a possible FTO issue which I know you would have a great opinion on, if you are interested in creating it. It's making the javascript files names consistent... #9113 (comment) would you be interested in creating FTOs for it?

I do have an opinion on that!!! We need to change it. And that's a perfect FTO issue! I'll make some of those!

**Comment by RuthNjeri** [comment]
Great! The names should also be updated in the `config/initializers/assets.rb` assets.rb file, I find that it is something that might be missed with all the Rails magic happening....

**Comment by cesswairimu** [comment]
Hey all, would anyone like to open this week's check-in?

**Comment by RuthNjeri** [comment]
Hi @cesswairimu I've created one [here ](https://github.com/publiclab/plots2/issues/9158)
